util/src/entities/BaseClass.ts

```diff
@@ -1,35 +1,24 @@
 import "reflect-metadata";
-import { BaseEntity, EntityMetadata, FindConditions, ObjectIdColumn, PrimaryColumn } from "typeorm";
+import { BaseEntity, EntityMetadata, ObjectIdColumn, PrimaryColumn, FindOptionsWhere, Generated, SaveOptions } from "typeorm";
 import { Snowflake } from "../util/Snowflake";
-import "missing-native-js-functions";
 
 export class BaseClassWithoutId extends BaseEntity {
 	constructor(props?: any) {
 		super();
-		this.assign(props);
-	}
-
-	private get construct(): any {
-		return this.constructor;
-	}
-
-	private get metadata() {
-		return this.construct.getRepository().metadata as EntityMetadata;
+		if(props != undefined && props != null && Object.keys(props).length > 0)
+			this.assign(props);
 	}
 
 	assign(props: any = {}) {
```

Comment:
as mentioned in the discord, this function - as well as toJSON - should not be removed. Object.assign doesn't do deep copy

Reply:
Replaced with a modified version of TypeORM's implementation.